CPTextView/Frameworks/TextView/CPTypesetter.j

```diff
@@ -352,26 +381,25 @@ var CPSystemTypesetterFactory;
                 else
                     rangeWidth += 28;   //FIXME
             }  // fallthrough intentional
-            case  ' ':
+            case 32: // ' '
                 wrapRange = CPMakeRangeCopy(lineRange);
                 wrapRange._height = _lineHeight;
                 wrapRange._base = _lineBase;
                 wrapWidth = rangeWidth;
                 break;
+            case 10:
+            case 13:
+                isNewline = YES;
+                break;
             default:
-                if (_isNewlineCharacter(currentChar))
-                {
-                    isNewline = YES;
-                }
         }
 
-        var advancement = CPMakeSize(rangeWidth - prevRangeWidth, ascent);
-        advancement.descent=descent;
+        var advancement = {width: rangeWidth - prevRangeWidth, height: ascent, descent: descent};
         advancements.push(advancement);
 
         prevRangeWidth = _lineWidth = rangeWidth;
 
-        if (lineOrigin.x + rangeWidth > containerSize.width)
+        if (lineOrigin.x + rangeWidth > containerSizeWidth)
```

Comment:
is this really relevant? i thought that references are handled very efficiently in modern browsers (we discussed that some time ago on the list iirc, in the context of eliminating the c #defines)